### PR TITLE
All my changes from the kom rework.

### PR DESCRIPTION
**Pull request by AraHaan**
## Change Description

All of my changes to the kom rework as well as others as well.

## Debug information

- Els_kom version: ``1.4.9.8``
- OS: ``Windows 7 Ultimate``
- OS Arch: ``x64``
- OS Build: ``7601``

### CODE REVIEW

**Comment by AraHaan** [comment]
ok, made this PR so I dont have to later, however dont review or merge yet until I complete the unpacker and packer code and fix the overflow and crashes at various parts of the unpacker.

File: Plugins/komplugins/Komv4_plugin.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+{
+    public class Komv4_plugin : Els_kom_Core.interfaces.IKomPlugin
+    {
+        public string KOMHeaderString => "KOG GC TEAM MASSFILE V.0.4.";
+        public int SupportedKOMVersion => 4;
+
+        public void Pack(string in_path, string out_path, string KOMFileName)
+        {
+            // not implemented yet due to lack of packing information on v4 koms.
+            throw new System.NotImplementedException();
```
Comment (CatGirlsAreLife) [inline Plugins/komplugins/Komv4_plugin.cs]: This is still not implemented. Need to figure out implementing this packing method.

File: Plugins/komplugins/Komv3_plugin.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+{
+    public class Komv3_plugin : Els_kom_Core.interfaces.IKomPlugin
+    {
+        public string KOMHeaderString => "KOG GC TEAM MASSFILE V.0.3.";
+        public int SupportedKOMVersion => 3;
+
+        public void Pack(string in_path, string out_path, string KOMFileName)
+        {
+            // not implemented yet due to lack of packing information on v3 koms.
+            throw new System.NotImplementedException();
```
Comment (CatGirlsAreLife) [inline Plugins/komplugins/Komv3_plugin.cs]: See review on kom v4 plugin.

File: Els_kom_Core/interfaces/ICallbackPlugin.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2014-2018, Els_kom org.
+// https://github.com/Elskom/
+// All rights reserved.
+// license: MIT, see LICENSE for more details.
+
+/* 0x01050000 = v1.5.0.0 */
```
Comment (CatGirlsAreLife) [inline Els_kom_Core/interfaces/ICallbackPlugin.cs]: I think the callback plugins should be in 1.4.9.8 for those who want to get a kom spoofing plugin for Elsword or Grand Chase sooner than 1.5.0.0.

**Pull request closed without merging**